apps/evm/src/components/Dropdown/index.tsx

```diff
@@ -0,0 +1,208 @@
+    const handleToggleDropdown = () => setIsDropdownOpened(!isDropdownOpened);
+
+    const isMdOrUp = useBreakpointUp('md');
+
+    const handleChange = (newValue: DropdownOption<TValue>['value']) => {
+      if (onChange) {
+        onChange(newValue);
+      }
+      setIsDropdownOpened(false);
+    };
+
+    const buttonSizeClasses = useMemo(() => {
+      if (size === 'large') {
+        return cn('px-4 h-14');
+      }
+
+      if (size === 'medium') {
+        return cn('px-4 h-12');
+      }
+
+      return cn('px-3 h-10');
+    }, [size]);
+
+    const optionsDom = (
+      <>
+        {options.map(option => (
+          <button
+            key={option.value}
+            onClick={() => handleChange(option.value)}
+            type="button"
+            className={cn(
+              'hover:bg-lightGrey active:bg-lightGrey flex min-w-full items-center justify-between py-3 text-left text-sm font-semibold',
+              buttonSizeClasses,
+            )}
+          >
+            <span className={cn('grow whitespace-nowrap')}>
+              {renderLabel({ label: option.label })}
+            </span>
+
+            <Icon
+              name="mark"
+              className={cn(
+                'text-green ml-4 w-5 h-5 shrink-0 opacity-0',
+                value === option.value && 'opacity-1',
+              )}
+            />
+          </button>
+        ))}
+      </>
+    );
+
+    return (
+      <>
+        <div className={cn(placeLabelToLeft && 'inline-flex items-center', className)}>
+          <input
+            ref={ref}
+            value={value}
+            className="hidden"
+            onChange={e => {
+              if (onChange) {
+                const formattedValue = (
+                  typeof value === 'number' ? +e.currentTarget.value : e.currentTarget.value
+                ) as TValue;
+
+                onChange(formattedValue);
+              }
+            }}
+            {...otherProps}
+          />
+
+          {!!label && (
+            <div className={cn(placeLabelToLeft ? 'mr-3 shrink-0' : 'mb-1')}>
+              <p className={cn('text-grey text-sm font-semibold')}>{label}</p>
+            </div>
+          )}
+
+          <div className="relative w-full">
+            {/* XS to MD backdrop */}
+            {isDropdownOpened && (
+              <div
+                className="fixed bottom-0 left-0 right-0 top-0 hidden md:block"
+                onClick={() => setIsDropdownOpened(false)}
+              />
+            )}
+
+            <Button
+              onClick={handleToggleDropdown}
+              className={cn(
+                'relative w-full',
+                getVariantClasses({ variant, isDropdownOpened }),
+                buttonSizeClasses,
+                buttonClassName,
+              )}
+              contentClassName={cn('w-full justify-between text-sm font-semibold')}
+            >
+              {children({ isDropdownOpened })}
+            </Button>
```

Comment:
Because the button applying horizontal padding, then it creates a gap on which we can click and close the dropdown without firing the `onClick` handler of the children:

https://github.com/user-attachments/assets/6ba921d0-b299-4dc6-abf3-4135c4103b2a

This issue would anyway be fixed by letting the children render their own button and handle closing/opening the dropdown menu.